comp/dogstatsd/listeners/uds_linux_test.go

```diff
@@ -37,7 +37,7 @@ func TestUDSPassCred(t *testing.T) {
 	pool := packets.NewPool(512)
 	poolManager := packets.NewPoolManager(pool)
 	config := fulfillDepsWithConfig(t, cfg)
-	s, err := NewUDSDatagramListener(nil, poolManager, nil, config, nil, optional.NewNoneOption[workloadmeta.Component]())
+	s, err := NewUDSDatagramListener(nil, poolManager, nil, config, nil, optional.NewNoneOption[workloadmeta.Component](), nil)
```

Comment:
NIT - instead of using `nil` we should either: 
1. if the component is expected to be optional, use an optional (as done with `workloadmeta.Component`)
2. use the mock (or real implementation). 

In this case I'd go with `2` since we always expect this component to exist at runtime.
I would consider updating `fulfillDepsWithConfig` to return multiple components (the config, and the pidmap component). 
 
 (same for the other tests)